tests/kernel/critical/src/Makefile

```diff
@@ -1,4 +1,3 @@
 ccflags-y += -I${ZEPHYR_BASE}/tests/include
 include $(ZEPHYR_BASE)/tests/Makefile.test
-
-obj-y = critical.o
+obj-y = main.o test_critical_context.o
```

Comment:
I don't really get it why we need to split the test definition and tests into different files.